deployability/modules/allocation/vagrant/instance.py

```diff
@@ -18,17 +19,30 @@ class VagrantInstance(Instance):
         path (str or Path): Directory where instance data is stored.
         identifier (str): Identifier of the instance.
         credentials (VagrantCredentials): Vagrant credentials object.
+        host_identifier (str, optional): The host for the instance. Defaults to None.
+        host_instance_dir (str | Path, optional): The remote directory of the instance. Defaults to None.
+        ssh_port (str): SSH port of the instance.
+        macos_host_parameters (dict): Parameters of the remote host.
     """
-    def __init__(self, path: str | Path, identifier: str, credentials: VagrantCredentials = None) -> None:
+    def __init__(self, instance_parameters: InstancePayload, credentials: VagrantCredentials = None) -> None:
         """
         Initializes a VagrantInstance.
 
         Args:
-            path (str | Path): The path of the instance.
-            identifier (str): The identifier of the instance.
+            instance_parameters (InstancePayload): The parameters of the instance.
             credentials (VagrantCredentials, optional): The credentials of the instance. Defaults to None.
         """
-        super().__init__(path, identifier, credentials)
+        super().__init__(instance_parameters, credentials)
+        self.path: Path = Path(instance_parameters.path)
+        self.identifier: str = instance_parameters.identifier
+        self.credentials: VagrantCredentials = credentials
+        self.host_identifier: str = instance_parameters.host_identifier
+        self.host_instance_dir: str | Path = instance_parameters.host_instance_dir
+        self.ssh_port: str = instance_parameters.ssh_port
+        self.macos_host_parameters: dict = instance_parameters.macos_host_parameters
+        self.arch: str = instance_parameters.arch
+        self.platform: str = instance_parameters.platform
+        self.user: str = instance_parameters.user
```

Comment:
There are some vars that are being defined here and in the inherited `Instance` class, thats not required, the vars are already defined in `Instance`, here you only have to configure this class specific attributes, anyway if its necessary you can override the variable, but I think it is not the case.

Reply:
Done eb2e2d830